api/geyser/src/main/java/org/geysermc/geyser/api/item/custom/CustomItemRegistrationTypes.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019-2022 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.item.custom;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.geysermc.geyser.api.GeyserApi;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+/**
+ * This class represents the different types of custom item registration
+ */
+public interface CustomItemRegistrationTypes {
+    @NonNull Optional<Boolean> unbreaking();
+
+    @NonNull OptionalInt customModelData();
+
+    @NonNull OptionalInt damagePredicate();
```

Comment:
Methods above are missing docs.

Also, we aren't using `Optional` anywhere else in the API so it may be better to just make these normal ints with a default value of 0 to better fit the design throughout the rest of the API.

Reply:
https://github.com/GeyserMC/Geyser/pull/2822#discussion_r863192752 I'll put it back...